### PR TITLE
add uploadFile and a path tag

### PR DESCRIPTION
**Pull request by gogolxdong**
Added uploadFile with progress and a path tag of VNodeKind which uses in svg

### CODE REVIEW

File: karax/kajax.nim [hunk 1]
```diff
@@ -4,6 +4,58 @@
...
+  FormData* {.importc.} = JsObject
+
+  HttpRequest* {.importc.} = ref object
+  ThisObj {.importc.} = ref object
+    readyState, status: int
+    responseText, statusText: cstring
+
+  XMLHttpRequestUpload* {.importc.} = JsObject
+
+var this* {.importc: "this".}: ThisObj
```
Comment (Araq) [inline karax/kajax.nim]: Don't export this variable, it's hacky.

File: karax/kajax.nim [hunk 2]
```diff
@@ -4,6 +4,58 @@
...
+proc statechange*(r: HttpRequest; cb: proc()) {.importcpp: "#.onreadystatechange = #".}
+proc send*(r: HttpRequest; data: cstring) {.importcpp: "#.send(#)".}
+proc send*(r: HttpRequest, data: Blob) {.importcpp: "#.send(#)".}
+proc open*(r: HttpRequest; meth, url: cstring; async: bool) {.importcpp: "#.open(@)".}
+proc newRequest*(): HttpRequest {.importcpp: "new XMLHttpRequest(@)".}
+
+proc uploadFile*(url: cstring, file: Blob, onprogress :proc(data: ProgressEvent), 
+                 headers: openarray[(cstring, cstring)] = []) = 
+  proc contWrapper(httpStatus: int; response: cstring) =
+    echo response
```
Comment (Araq) [inline karax/kajax.nim]: Debug ``echo`` here? Bad in a library.

Comment (gogolxdong) [reply karax/kajax.nim]: Added a cont to have the processing proc to be passed in customly.

File: karax/kajax.nim (hunk 1 above)
Comment (gogolxdong) [reply karax/kajax.nim]: oh, I removed it inside proc where uses it, not sure it's the same with what you image.

**Pull request merged**